lib/applications/resolvers.ts

```diff
@@ -713,3 +717,129 @@ export const generateApplicationsReport: Resolver = async (_, args, { prisma })
     ok: true,
   };
 };
+
+export const generatePermitHoldersReport: Resolver = async (_, args, { prisma }) => {
+  const {
+    input: { startDate, endDate, columns },
+  } = args;
+
+  const columnsSet = new Set(columns);
+
+  const applications = await prisma.application.findMany({
+    where: {
+      createdAt: {
+        gte: startDate,
+        lte: endDate,
+      },
+    },
+    select: {
+      rcdUserId: columnsSet.has(PermitHoldersReportColumn.UserId),
+      firstName: columnsSet.has(PermitHoldersReportColumn.ApplicantName),
+      middleName: columnsSet.has(PermitHoldersReportColumn.ApplicantName),
+      lastName: columnsSet.has(PermitHoldersReportColumn.ApplicantName),
+      dateOfBirth: columnsSet.has(PermitHoldersReportColumn.ApplicantDateOfBirth),
+      addressLine1: columnsSet.has(PermitHoldersReportColumn.HomeAddress),
+      addressLine2: columnsSet.has(PermitHoldersReportColumn.HomeAddress),
+      city: columnsSet.has(PermitHoldersReportColumn.HomeAddress),
+      province: columnsSet.has(PermitHoldersReportColumn.HomeAddress),
+      postalCode: columnsSet.has(PermitHoldersReportColumn.HomeAddress),
+      email: columnsSet.has(PermitHoldersReportColumn.Email),
+      phone: columnsSet.has(PermitHoldersReportColumn.PhoneNumber),
+      guardianFirstName: columnsSet.has(PermitHoldersReportColumn.GuardianPoaName),
+      guardianMiddleName: columnsSet.has(PermitHoldersReportColumn.GuardianPoaName),
+      guardianLastName: columnsSet.has(PermitHoldersReportColumn.GuardianPoaName),
+      guardianRelationship: columnsSet.has(PermitHoldersReportColumn.GuardianPoaRelation),
+      guardianAddressLine1: columnsSet.has(PermitHoldersReportColumn.GuardianPoaAddress),
+      guardianAddressLine2: columnsSet.has(PermitHoldersReportColumn.GuardianPoaAddress),
+      guardianPostalCode: columnsSet.has(PermitHoldersReportColumn.GuardianPoaAddress),
+      guardianCity: columnsSet.has(PermitHoldersReportColumn.GuardianPoaAddress),
+      guardianProvince: columnsSet.has(PermitHoldersReportColumn.GuardianPoaAddress),
+      permit: {
+        select: {
+          rcdPermitId: true,
+          // NOTETOSELF: Is user status whether the permit is active,
+          // or is it the ApplicationStatus (taken from ApplicationProcessing),
+          // or is it the ApplicantStatus (taken from Applicant)
+          active: columnsSet.has(PermitHoldersReportColumn.UserStatus),
```

Comment:
@OustanDing I wasn't sure what User Status maps to in the db. Would you happen to know?
![image](https://user-images.githubusercontent.com/51224641/146285270-76343a55-6ed1-4202-a925-9b619161617e.png)